src/main/kotlin/com/workos/auditlogs/AuditLogsApi.kt

```diff
@@ -0,0 +1,179 @@
+      private var context: Context? = null
+      private var version: Int? = 1
+      private var metadata: Map<String, String>? = null
+
+      /**
+       * Sets the action.
+       */
+      fun action(value: String) = apply { action = value }
+
+      /**
+       * Sets the actor.
+       */
+      @JvmOverloads
+      fun actor(
+        id: String,
+        type: String,
+        name: String? = null,
+        metadata: Map<String, String>? = null,
+      ) = apply { actor = Actor(id, type, name, metadata) }
+
+      /**
+       * Sets the context.
+       */
+      @JvmOverloads
+      fun context(
+        location: String,
+        userAgent: String? = null,
+      ) = apply { context = Context(location, userAgent) }
+
+      /**
+       * Sets the occurred at date.
+       */
+      fun occurredAt(value: Date) = apply { occurredAt = value }
+
+      /**
+       * Adds a target to the targest list.
+       */
+      @JvmOverloads
+      fun target(
+        id: String,
+        type: String,
+        name: String? = null,
+        metadata: Map<String, String>? = null,
+      ) = apply { targets.add(Target(id, type, name, metadata)) }
+
+      /**
+       * Sets the context.
+       */
+      fun version(value: Int) = apply { version = value }
+
+      /**
+       * Sets the metadata.
+       */
+      fun metadata(value: Map<String, String>) = apply { metadata = value }
+
+      /**
+       * Creates a [EnrollFactorOptions] with the given builder parameters.
+       */
+      fun build(): CreateAuditLogEventOptions {
+        if (action == null) {
+          throw IllegalArgumentException("An action must be provided")
+        }
+
+        if (actor == null) {
+          throw IllegalArgumentException("An actor must be provided")
+        }
+
+        if (context == null) {
+          throw IllegalArgumentException("A context must be provided")
+        }
+
+        if (context == null) {
+          throw IllegalArgumentException("A context must be provided")
+        }
+
+        if (occurredAt == null) {
+          throw IllegalArgumentException("An occurred at date must be provided")
+        }
+
+        if (targets.isEmpty()) {
+          throw IllegalArgumentException("At least one target must be provided")
+        }
+
+        return CreateAuditLogEventOptions(
+          action = action!!,
+          actor = actor!!,
+          context = context!!,
+          occurredAt = occurredAt!!,
+          targets = targets,
+          version = version,
+          metadata = metadata,
+        )
+      }
+    }
+
+    /**
+     * @suppress
+     */
+    companion object {
+      @JvmStatic
+      fun builder(): CreateAuditLogEventOptionsBuilder {
+        return CreateAuditLogEventOptionsBuilder()
+      }
+    }
+  }
+
+  /**
+   * Emits an Audit Log event.
+   */
+  fun createEvent(organizationId: String, createAuditLogEventOptions: CreateAuditLogEventOptions) {
```

Comment:
perhaps support `idempotencyKey` like we do in the SDK to prevent adding duplicate events

Reply:
Good call, updating.